api/v1/managementcluster_types.go

```diff
@@ -27,6 +27,32 @@ type ManagementClusterSpec struct {
 	// Valid examples are: "0.0.0.0:31000", "example.com:32000", "[::1]:32500"
 	// +optional
 	Address string `json:"address,omitempty"`
+
+	// TLS provides options for configuring how Managed Clusters can establish an mTLS connection with the Management Cluster.
+	// +optional
+	TLS *TLS `json:"tls,omitempty"`
+}
+
+type TLS struct {
+	// SecretName indicates the name of the secret in the tigera-operator namespace that contains the private key and certificate that the management cluster uses when it listens for incoming connections.
+	//
+	// When set to tigera-management-cluster-connection voltron will use the same cert bundle which Guardian client certs are signed with.
+	//
+	// When set to manager-tls, voltron will use the same cert bundle which Manager UI is served with.
+	// This cert bundle must be a publicly signed cert created by the user.
+	// Note that Tigera Operator will generate a self-signed manager-tls cert if one does not exist,
+	// and use of that cert will result in Guardian being unable to verify Voltron's identity.
+	//
+	// If changed on a running cluster with connected managed clusters, all managed clusters will disconnect as they will no longer be able to verify Voltron's identity.
+	// To reconnect existing managed clusters, change the tls.ca of the  managed clusters' ManagementClusterConnection resource.
+	//
+	// One of: tigera-management-cluster-connection, manager-tls
+	//
+	// Default: tigera-management-cluster-connection
+	//
+	// +kubebuilder:validation:Enum=tigera-management-cluster-connection;manager-tls
+	// +optional
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
Not sure about the context for this one, but as far as API design goes there are existing constructs for this.

You might want to look into: `v1.LocalObjectReference`